app/src/main/java/com/example/mycalculator/MainActivity.java

```diff
@@ -2,13 +2,99 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.annotation.SuppressLint;
 import android.os.Bundle;
+import android.view.View;
+
+import com.example.mycalculator.databinding.ActivityMainBinding;
+import com.example.mycalculator.mvp.CalculatorContract;
+import com.example.mycalculator.mvp.CalculatorModelImpl;
+import com.example.mycalculator.mvp.CalculatorPresenterImpl;
+import com.example.mycalculator.mvp.CalculatorViewImpl;
 
 public class MainActivity extends AppCompatActivity {
 
+    private ActivityMainBinding binding;
+    private CalculatorContract.CalculatorPresenterContract presenter;
+
+    @SuppressLint("WrongViewCast")
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
+        binding = ActivityMainBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        presenter = new CalculatorPresenterImpl(new CalculatorModelImpl(), new CalculatorViewImpl(this, binding));
+
+        setListeners();
+    }
+
+    private void setListeners() {
+        binding.buttonNumberZero.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberZero.getText().toString());
```

Comment:
as you call the **.toString()** the String cast is not necessary anymore. Please remove it

Reply:
Done. Thx!